src/openms/source/CHEMISTRY/ModificationsDB.cpp

```diff
@@ -257,6 +289,8 @@ namespace OpenMS
   {
     double min_error = max_error;
     const ResidueModification* mod = nullptr;
+    char res = '?'; // empty
+    if (!residue.empty()) res = residue[0];
```

Comment:
not benchmarked at all, but to avoid the (costly) branch you could do:
```
char candidates[2] = { residue[0], '?' };
const char res = candidates[residue.empty()];
```

`residue[0]` should not be UB since the empty string is guaranteed to end in `\0' as of C++11.

Reply:
I did the benchmarking on godbolt: https://godbolt.org/z/zFppCc and it looks like on (gcc): 5 instructions for my solution, 8 instructions for yours. I suggest we stick to the one there is